server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -94,6 +99,8 @@ public DecommissionService(
             CLUSTER_ROUTING_ALLOCATION_AWARENESS_FORCE_GROUP_SETTING,
             this::setForcedAwarenessAttributes
         );
+        maxVotingConfigExclusions = MAXIMUM_VOTING_CONFIG_EXCLUSIONS_SETTING.get(settings);
+        clusterSettings.addSettingsUpdateConsumer(MAXIMUM_VOTING_CONFIG_EXCLUSIONS_SETTING, this::setMaxVotingConfigExclusions);
```

Comment:
`DecommissionService` is a consumer to this dynamic setting's value. We need this here since now we resolve the nodes to excluded and update the state directly, we need to check if the exclusion that we are adding is not breaching the max limit set by the user. This is the same behaviour in the corresponding transport action of voting config exclusion

Reply:
I would suggest keeping this state on the `VotingConfigExclusionsHelper` and avoid making the class static. That should help move any state centrally. Also please re-name to `VotingConfigExclusionsExecutor`

Reply:
`VotingConfigExclusionsHelper` is just a util class which just exposes a few public methods to update cluster state blob given in the method param. Other than this there's no use of it. Do you see a problem if service becomes a consumer to a particular setting?

Reply:
Do we absolutely need a service class here? As I don't see that we need to maintain any other state other than a dynamic setting here and rest all are static methods. We can't send a state update call in this service too, as we just need these static helpers to update the cluster state blob and return it as is. Respective services (eg Decommission and VotingConfigExclusion) here will execute the state update calls.
Decommission service is just a consumer to the dynamic setting (we have many other services which does that).